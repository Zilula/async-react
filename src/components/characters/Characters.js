import React, { Component, Fragment } from 'react';
import Character from './Character';
import { getCharacters } from '../services/rickAndMortyApi';


export default class App extends Component {
    state = {
        characters: []
    };

    componentDidMount() {
        getCharacters()
            .then(chars => {
                this.setState({ characters: chars.results });
            });
    }

    render() {
        const characterList = this.state.characters.map(char => {
            return (
                <Character character={char} key={char.name}/>
            );
        });
        return (
            <Fragment>
                <ul>
                    {characterList}
                </ul>
            </Fragment>
        );
    }
}

