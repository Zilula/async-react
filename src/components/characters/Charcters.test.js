import React from 'react';
import renderer from 'react-test-renderer';
import Characters from './Characters';

describe('does things', () => {
    it('matches a snapchat', () => {
        const tree = renderer.create(
            <Characters/>
        );
        expect(tree).toMatchSnapshot();
    });
});