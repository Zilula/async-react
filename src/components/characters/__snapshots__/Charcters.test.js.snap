// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does things matches a snapchat 1`] = `
<ul>
  <li>
    <p>
       
      Rick Sanchez
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Morty Smith
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Summer Smith
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Female
       
    </p>
  </li>
  <li>
    <p>
       
      Beth Smith
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Female
       
    </p>
  </li>
  <li>
    <p>
       
      Jerry Smith
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Abadango Cluster Princess
       
    </p>
    <p>
       
      Alien
       
    </p>
    <p>
       
      Female
       
    </p>
  </li>
  <li>
    <p>
       
      Abradolf Lincler
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Adjudicator Rick
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Agency Director
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Alan Rails
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Albert Einstein
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Alexander
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Alien Googah
       
    </p>
    <p>
       
      Alien
       
    </p>
    <p>
       
      unknown
       
    </p>
  </li>
  <li>
    <p>
       
      Alien Morty
       
    </p>
    <p>
       
      Alien
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Alien Rick
       
    </p>
    <p>
       
      Alien
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Amish Cyborg
       
    </p>
    <p>
       
      Alien
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Annie
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Female
       
    </p>
  </li>
  <li>
    <p>
       
      Antenna Morty
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Antenna Rick
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
  <li>
    <p>
       
      Ants in my Eyes Johnson
       
    </p>
    <p>
       
      Human
       
    </p>
    <p>
       
      Male
       
    </p>
  </li>
</ul>
`;
